tests/src/test/scala/ha/ShootComponentsTests.scala

```diff
@@ -305,9 +305,8 @@ trait ShootComponentUtils {
     def file(path: String) = Try(new File(path)).filter(_.exists).map(_.getAbsolutePath).toOption
 
     val docker = (file("/usr/bin/docker") orElse file("/usr/local/bin/docker")).getOrElse("docker")
-    val dockerPort = WhiskProperties.getProperty(WhiskConfig.dockerPort)
 
-    Seq(docker, "--host", host + ":" + dockerPort, cmd, component)
+    Seq(docker, cmd, component)
```

Comment:
How is the test supposed to work if there is more than one controller running and the controllers run on different systems?

The original test code determines which host is running the first controller and would ask the remote host's Docker daemon to start, stop and restart the controller container. With the change, the test permanently fails in our test environment that uses multiple systems in parallel.

Reply:
I just want to completely disable remote Docker API from the code as we don't use it anymore in our normal use cases.
It seems remote docker API is only required for test cases.

We can selectively enable/disable remote API based on the deployment type such as 
 production or unit tests.
But it depends on the OW operator, there is still a possibility that Docker remote API is enabled without consciousness. And it could induce a security breach.

How about making this test rely on `ssh` command instead?
Then what we need is just to make ssh connection available between target hosts and machine which runs tests.
(Generally, tests are executed in build machines such as jenkins, this requirement would be highly likely already met.)

Reply:
I actually opt for keeping the current approach with running Docker commands against remote systems. From a security perspective, it's a matter of how you run your Docker daemon. In test environments, you can open a TCP listening socket to accept connections from remote systems - and on production systems, you can just disable the TCP listening socket. The rest is how you configure openwhisk.

Today, the `DockerClientWithFileAccess()` is created with an empty `host` parameter anyway such that the local Docker daemon BSD stream socket is used.

Reply:
Make sense.
I just wanted to clean up this configuration as well: https://github.com/apache/incubator-openwhisk/pull/4430/files#diff-85b6ff9e51a5005c641ac4690b975b2cL55
But that's also a viable option.
I will work on it.